nox/_option_set.py

```diff
@@ -83,7 +83,11 @@ def __init__(
         noxfile: bool = False,
         merge_func: Callable[[Namespace, Namespace], Any] | None = None,
         finalizer_func: Callable[[Any, Namespace], Any] | None = None,
-        default: Any | Callable[[], Any] = None,
+        default: bool
+        | str
+        | None
+        | list[str]
+        | Callable[[], bool | str | None | list[str]] = None,
```

Comment:
This should probably be generic eventually - `Option[str]` would hold a str, etc. But didn't look into it.